aggregators/converter.go

```diff
@@ -989,26 +990,56 @@ func overflowSpanMetricsToAPMEvent(
 }
 
 func marshalEventGlobalLabels(e *modelpb.APMEvent) ([]byte, error) {
-	var gl GlobalLabels
+	if len(e.Labels) == 0 && len(e.NumericLabels) == 0 {
+		return nil, nil
+	}
+
+	pb := aggregationpb.GlobalLabelsFromVTPool()
+	defer pb.ReturnToVTPool()
+
+	// Keys must be sorted to ensure wire formats are deterministically generated and strings are directly comparable
+	// i.e. Protobuf formats are equal if and only if the structs are equal
 	for k, v := range e.Labels {
 		if !v.Global {
 			continue
 		}
-		if (&gl).Labels == nil {
-			(&gl).Labels = make(modelpb.Labels)
+		if len(pb.Labels) == cap(pb.Labels) {
+			pb.Labels = append(pb.Labels, &aggregationpb.Label{})
+		} else {
+			pb.Labels = pb.Labels[:len(pb.Labels)+1]
+			if pb.Labels[len(pb.Labels)-1] == nil {
+				pb.Labels[len(pb.Labels)-1] = &aggregationpb.Label{}
+			}
 		}
-		gl.Labels[k] = v
+		pb.Labels[len(pb.Labels)-1].Key = k
+		pb.Labels[len(pb.Labels)-1].Value = v.Value
+		pb.Labels[len(pb.Labels)-1].Values = v.Values
```

Comment:
[For reviewer] I hate myself for writing this reslicing code again, but this should perform better than looping over the map twice. This time since we don't know the resulting length beforehand, we can't grow and reslice at once before the loop. The equiv vtprotobuf code is [here](https://github.com/planetscale/vtprotobuf/blob/main/features/unmarshal/unmarshal.go#L581).